io.cpp

```diff
@@ -40,6 +40,32 @@ Colour ansiTable[8]={black,red,green,yellow,blue,magenta,cyan,white};
 unsigned int screencolours;
 
 unordered_map<string,string> tputcache;
+enum KEY_ACTION{
+			KEY_NULL = 0,       /* NULL */
+			CTRL_C = 3,         /* Ctrl-c */
```

Comment:
Many of them. They are some specialty keys that can get returned by the `editorReadKey` function. Not all are being used right now.

As for my previous comment, yes, I do realize now that the 'passing by reference' comment was about something different.  I don't remember why I added that comment in the first place, but I can remove it if you'd like.